tools/clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -7596,6 +7596,10 @@ def err_hlsl_unsupported_buffer_packoffset : Error<
   "packoffset is only allowed within a constant buffer, not on the constant buffer declaration">;
 def err_hlsl_unsupported_buffer_slot_target_specific : Error<
   "user defined constant buffer slots cannot be target specific">;
+def err_hlsl_unsupported_typedbuffer_template_parameter : Error<
+  "Elements of typed buffers and textures must be scalars or vectors">;
```

Comment:
```suggestion
  "elements of typed buffers and textures must be scalars or vectors">;
```

Reply:
Applied in new commit along with all the test changes.